### PR TITLE
feat: plonk profiling test for recursion circuit

### PR DESCRIPTION
**Pull request by erabinov**

### CODE REVIEW

File: prover/src/new_circuit_tests.rs [hunk 1]
```diff
@@ -0,0 +1,154 @@
+#[cfg(test)]
```
Comment (jtguibas) [inline prover/src/new_circuit_tests.rs:1]: this is not the right folder to put this test in, please put it inside core-v2 or a v2 folder

File: prover/src/new_circuit_tests.rs [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+    use std::{borrow::Borrow, iter::once};
+
+    use p3_baby_bear::{BabyBear, DiffusionMatrixBabyBear};
+    use p3_field::extension::BinomialExtensionField;
+    use sp1_core::{
+        stark::StarkMachine,
+        utils::{log2_strict_usize, run_test_machine, setup_logger, BabyBearPoseidon2Inner},
+    };
+    use sp1_recursion_compiler::{config::OuterConfig, ir::Witness};
+    // use sp1_recursion_compiler::{config::OuterConfig, constraints::ConstraintCompiler, ir::Felt};
```
Comment (jtguibas) [inline prover/src/new_circuit_tests.rs:12]: this file needs to be a lot cleaner, lots of random uncommented stuff

File: recursion/circuit/src/stark.rs [hunk 3]
```diff
@@ -361,16 +363,142 @@ pub fn build_wrap_circuit(
     backend.emit(builder.operations)
 }
 
+/// A function to build the Plonk constraints for a proof and a verifying key. The main difference
+/// between this function and the "old" one is that we use the RecursionAir machine from core-v2.
+pub fn build_wrap_circuit_new<F, const DEGREE: usize, const COL_PADDING: usize>(
```
Comment (jtguibas) [inline recursion/circuit/src/stark.rs:368]: build_wrap_circuit_v2?

Comment (jtguibas) [reply recursion/circuit/src/stark.rs:368]: also i feel like the comment can be made "more professional" / "concise".

/// A function to build the circuit for the wrap layer using core-v2.

File: prover/src/new_circuit_tests.rs (hunk 1 above)
Comment (erabinov) [reply prover/src/new_circuit_tests.rs]: I made a `circuit_v2` folder and moved the `build_wrap_circuit_v2` there also.

**Review by jtguibas — APPROVED** [review]

**Pull request merged**